shoop/xtheme/theme.py

```diff
@@ -6,22 +6,137 @@
 # This source code is licensed under the AGPLv3 license found in the
 # LICENSE file in the root directory of this source tree.
 from contextlib import contextmanager
-from shoop.apps.provides import get_provide_objects
+from shoop.apps.provides import get_provide_objects, get_identifier_to_object_map
+from shoop.xtheme.models import ThemeSettings
 
 
+# TODO: Document how to create Xthemes
+
 class Theme(object):
+    """
+    Base class for all Xtheme themes.
+
+    This class does not directly correspond to a database object;
+    it's used in the rendering, etc. process.
+
+    It does, however, act as a container for a `ThemeSettings` object
+    that contains the actual persisted settings etc.
+    """
+
     # The identifier for this theme. Used for theme selection.
+    # Must be set in actual themes!
     identifier = None
 
+    # The printable name for this theme.
+    name = ""
+
+    # The author for this theme, for posterity.
+    author = ""
+
     # Directory prefix for this theme's template files.
     # If `None`, the `identifier` is used instead
     template_dir = None
 
     # List of global placeholder names.
     # TODO: (These could be ignored in per-view editing, or something?)
+    # TODO: Document this
     global_placeholders = []
 
-    global_variables = []
+    # List of (name, FormField) pairs for theme configuration.
+    # This might not be used if you override `get_configuration_form`.
+    # TODO: Document this
+    fields = []
+
+    def __init__(self, settings_obj=None):
+        """
+        Initialize this theme, with an optional `ThemeSettings` object
+
+        :param settings_obj: A theme settings object for this theme, if one exists.
+                             Passing this in will avoid extraneous database queries.
+        :type settings_obj: ThemeSettings|None
+        """
+        self._settings_obj = None
+        if settings_obj and settings_obj.theme_identifier == self.identifier:  # fine, let's accept that
+            self._settings_obj = settings_obj
+
+    @property
+    def settings_obj(self):
```

Comment:
I would like `def settings(self)` even though it's different from django `settings`